internal/pbutil/shallow_equal.go

```diff
@@ -0,0 +1,54 @@
+package pbutil
+
+import (
+	"errors"
+	"reflect"
+)
+
+// ShallowEqual compares two things in a shallow (not deep like reflect.DeepEqual) way.
+// if they're pointers, they're dereferenced first before proceeding.
+// if they're structs, it does Value.Equal() for all visible fields whose types are .Comparable(), and ignores any other fields.
+// otherwise, if they're some other type that is .Comparable(), it just does Value.Equal() and hopes for the best.
+// if none of the above apply, it returns false.
+// superceding any of the above, if an error occurs, false will be returned along with the error.
+// note that this method is different from reflect.Equal in that it does not care if the types are the same, only the kinds.
+func ShallowEqual(a, b any) (bool, error) {
```

Comment:
`go-cmp` package has similar comparison logic, can we use it instead? https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal

Reply:
`cmp` is not suitable for production, only for tests; this is called out in their documentation and it's also something that was being used by this repo with an explicit callout to replace it with something production-ready.